modular_skyrat/modules/conveyor_sorter/code/conveyor_sorter.dm

```diff
@@ -86,20 +88,16 @@
 	var/user_choice = tgui_input_list(user, "Choose which direction to sort to!", "Direction choice", list("North", "East", "South", "West"))
 	if(!user_choice)
 		return ..()
-	switch(user_choice)
-		if("North")
-			setDir(NORTH)
-		if("East")
-			setDir(EAST)
-		if("South")
-			setDir(SOUTH)
-		if("West")
-			setDir(WEST)
+	var/dir = text2dir(user_choice)
+	if(!dir)
+		return ..()
+	setDir(dir)
+
 	visible_message("[src] pings, updating its sorting direction!")
 	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
 
 /obj/effect/decal/cleanable/conveyor_sorter/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/conveyor_sorter))
+	if(istype(W, /obj/item/conveyor_sorter) || istype(W, /obj/item/conveyor_sorter/improved))
 		var/obj/item/conveyor_sorter/cs_item = W
```

Comment:
Replace `W` by `used_item` here please (I'd do a suggestion but GitHub is stupid and won't let me do it).